src/components/ShowOptions/ShowOptions.jsx

```diff
@@ -26,59 +26,62 @@ export const ShowOptions = ({ video }) => {
-              <i
-                className="bx bx-like"
-                onClick={() => {
-                  isAuthenticated
-                    ? addToLiked(video, dataDispatch)
-                    : toast.warning("Please login first!");
-                }}
-              ></i>
-            )}
-          </li>
-          <li className="show-items">
-            {inWatchLater ? (
-              <i
-                class="bx bxs-time-five"
-                onClick={() => removeFromWatchLater(video._id, dataDispatch)}
-              ></i>
-            ) : (
-              <i
-                className="bx bx-time-five"
-                onClick={() => {
-                  isAuthenticated
-                    ? addToWatchLater(video, dataDispatch)
-                    : toast.warning("Please login first!");
-                }}
-              ></i>
-            )}
-          </li>
-          <li
-            className="show-items"
-            onClick={(e) => {
-              if (isAuthenticated) {
-                e.stopPropagation();
-                setShowModal(true);
-              } else toast.warning("Please login first!");
-            }}
-          >
-            <i className="bx bx-list-plus"></i>
-          </li>
-        </ul>
-      )}
+      <div className="show-options-container ml-auto">
+        <i
+          className="bx bx-dots-vertical-rounded ml-auto icon"
+          onClick={() => setIsHidden(!isHidden)}
+        ></i>
+        {!isHidden && (
+          <ul className="show-list">
+            <li className="show-items">
+              {isLikedVideo ? (
+                <i
+                  class="bx bxs-like"
+                  onClick={() => removeFromLiked(video._id, dataDispatch)}
+                ></i>
+              ) : (
+                <i
+                  className="bx bx-like"
+                  onClick={() => {
+                    isAuthenticated
+                      ? addToLiked(video, dataDispatch)
+                      : toast.warning("Please login first!");
+                  }}
+                ></i>
+              )}
+            </li>
+            <li className="show-items">
+              {inWatchLater ? (
+                <i
+                  class="bx bxs-time-five"
+                  onClick={() => removeFromWatchLater(video._id, dataDispatch)}
+                ></i>
+              ) : (
+                <i
+                  className="bx bx-time-five"
+                  onClick={() => {
+                    isAuthenticated
+                      ? addToWatchLater(video, dataDispatch)
+                      : toast.warning("Please login first!");
+                  }}
+                ></i>
+              )}
+            </li>
+            <li
+              className="show-items"
+              onClick={(e) => {
+                if (isAuthenticated) {
+                  e.stopPropagation();
+                  setShowModal(true);
+                } else toast.warning("Please login first!");
```

Comment:
if the onclick is not a single task, bring it out as a function and call it there.

Reply:
Done